src/printer.js

```diff
@@ -1052,8 +1052,23 @@ function printExpression(path, options, print) {
             ? hardline
             : ""
         ]);
-      case "inline":
-        return join(literalline, node.raw.split("\n"));
+      case "inline": {
+        const printComments = comments =>
+          comments && comments.length
+            ? concat([
+                "<?php",
+                hardline,
+                join(hardline, comments.map(c => c.value)),
+                hardline,
+                "?>"
+              ])
+            : "";
+        return concat([
+          printComments(node.leadingComments),
+          join(literalline, node.raw.split("\n")),
+          printComments(node.comments)
+        ]);
+      }
```

Comment:
I think better move this logic in `printLines`, in code we should add `<?php` and `?>` only in `printLines`, otherwise it can break a lot of cases which is not tests right now

Reply:
Right - I'll try to understand how `printLines` is working :wink: